### PR TITLE
PT-1447: Better usability for the pedigree date picker; PT-2274: Date is now saved before blur

### PR DESCRIPTION
**Pull request by veronikaslc**
- Better behavior for year input; slight improvement to input character capturing

### CODE REVIEW

**Comment by veronikaslc** [comment]
The PR is based on the https://github.com/phenotips/phenotips/pull/1944.
For the convenience of codereviewing and rebasing on 2498 commits of master I have squashed commits in one.

File: components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js [hunk 1]
```diff
@@ -1,264 +1,401 @@
...
-      return this.daySelector.getElement();
+    onKeyPress : function($super, e) {
+      return $super(e, /[0-9]/, 2, true);
     },
 
-    getAvailableDays : function () {
-      var year = this.yearSelector.getSelectedValue() * 1;
-      var month = this.monthSelector.getSelectedValue() * 1;
+    onDateChange : function($super) {
+      $super();
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js]: We need to make sure this plays well with different date formats. I see the logic to make it `alwaysEnabled`, but somehow this is turned off when `yearInput.isFuzzy`. Maybe that is ok, need to think about that.

File: components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js [hunk 2]
```diff
@@ -1,264 +1,401 @@
...
+    insertInputs : function(){
+      this.inputFormat.toLocaleLowerCase().split("").forEach(function(identifier){
+        this.container.insert(this.inputs[identifier].getElement());
+      }.bind(this));
+    },
+    updateDateFromInput : function() {
+      var dateObject = JSON.parse(this._input.value);
+      if (dateObject["year"] != null) {
+        this.inputs.y.setValue(dateObject["year"]);
+      } else if (dateObject["decade"] != null) {
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js]: This is using an outdated date format I think, we no longer have a `decade` field, instead we have a `range` field

A few things need to be updated (here and in a few other places - at least everywhere there is aword "decade") to support current JSON Fuzzy Date format we use in both pedigree and PT

**Comment by veronikaslc** [comment]
postponing the PR, it was decided to make this datepicker a widget

**Pull request closed without merging**